src/stdatamodels/model_base.py

```diff
@@ -624,6 +624,13 @@ def to_fits(self, init, *args, **kwargs):
             with warnings.catch_warnings():
                 warnings.filterwarnings('ignore', message='Card is too long')
                 if self._no_asdf_extension:
+                    # For some old files that were written out before the
+                    # _no_asdf_extension existed, these will have an ASDF
+                    # extension, which may get passed along through extra_fits.
+                    # Avoid this.
+                    if "ASDF" in [hdu.name for hdu in ff._hdulist]:
```

Comment:
The HDUList class implements `__contains__` so we can just do `if "ASDF" in ff._hdulist` here?

Reply:
Nice catch.  I will change this here and in the test.